src/caustics/utils.py

```diff
@@ -115,6 +116,124 @@ def get_meshgrid(
     )
     return torch.meshgrid([xs, ys], indexing="xy")
 
+@lru_cache(maxsize=32)
+def _quad_table(n, p, dtype, device):
+    """
+    Generate a meshgrid for quadrature points using Legendre-Gauss quadrature.
+
+    Parameters
+    ----------
+    n : int
+        The number of quadrature points in each dimension.
+    p : torch.Tensor
+        The pixelscale.
+    dtype : torch.dtype
+        The desired data type of the tensor.
+    device : torch.device
+        The device on which to create the tensor.
+
+    Returns
+    -------
+    Tuple[torch.Tensor, torch.Tensor, torch.Tensor]
+        The generated meshgrid as a tuple of Tensors.
+    """
+    abscissa, weights = roots_legendre(n)
+
+    w = torch.tensor(weights, dtype=dtype, device=device)
+    a = p * torch.tensor(abscissa, dtype=dtype, device=device) / 2.0
+    X, Y = torch.meshgrid(a, a, indexing="xy")
+
+    W = torch.outer(w, w) / 4.0
+
+    X, Y = X.reshape(-1), Y.reshape(-1) # flatten
+    return X, Y, W.reshape(-1)
+
+def get_pixel_quad_integrator_grid(
```

Comment:
We should not name function `get_something`, instead we should just call the function `something`. This helps readability. Why not integrate this function into the `pixel_quad_integrator`?